uni/lib/view/schedule/widgets/schedule_slot.dart

```diff
@@ -104,21 +108,38 @@ class ScheduleSlot extends StatelessWidget {
 
 class SubjectButtonWidget extends StatelessWidget {
   const SubjectButtonWidget({required this.occurrId, super.key});
+
   final int occurrId;
+  CourseUnit correspondingCourseUnit(BuildContext context) {
+    final courseUnits =
+        Provider.of<ProfileProvider>(context, listen: false).state!.courseUnits;
+    final notFound =
+        CourseUnit(abbreviation: 'NF', name: 'not found', occurrId: 0);
+    final correspondingCourseUnit = courseUnits.firstWhere(
+      (courseUnit) => courseUnit.occurrId == occurrId,
+      orElse: () => notFound,
+    );
 
```

Comment:
Isn't there a way to pass down the courseUnit from the parent that build the schedule slot?

Reply:
Try to avoid sentinel values. Return null in case of error.